apps/src/lib/node/ledger/shell/finalize_block.rs

```diff
@@ -357,10 +391,70 @@ where
         tracing::info!("{}", stats);
         tracing::info!("{}", stats.format_tx_executed());
 
-        if new_epoch {
+        if update_for_tendermint {
             self.update_epoch(&mut response);
         }
 
+        // Read the block proposer of the previously committed block in storage
+        // (n-1 if we are in the process of finalizing n right now).
+        match read_last_block_proposer_address(&self.wl_storage)? {
+            Some(proposer_address) => {
+                println!("FOUND LAST BLOCK PROPOSER");
+                if new_epoch {
+                    self.apply_inflation(
+                        current_epoch,
+                        &proposer_address,
+                        &req.votes,
+                    )?;
+                } else {
+                    namada_proof_of_stake::log_block_rewards(
+                        &mut self.wl_storage,
+                        current_epoch,
+                        &proposer_address,
+                        &req.votes,
+                    )
+                    .unwrap();
+                }
```

Comment:
The call to `log_block_rewards` that is inside of `apply_inflation` uses the previous epoch rather than the `current_epoch`, so this suggestion would be incorrect. I'd say to leave as is if you don't have a strong preference.